src/utils/data-stream.ts

```diff
@@ -0,0 +1,48 @@
+import { Encoder } from './encoder.js';
+import { Readable } from 'readable-stream';
+
+/**
+ * Utility class for readable data stream, intentionally named to disambiguate from ReadableStream, readable-stream, Readable etc.
+ */
+export class DataStream {
+  /**
+   * Reads the entire readable stream given into array of bytes.
+   */
+  public static async toBytes(readableStream: Readable): Promise<Uint8Array> {
+    return new Promise((resolve, reject) => {
+      const chunks = [];
+      readableStream.on('data', chunk => {
```

Comment:
does `ReadableStream` actually dispatch any events?

why not instead call `read()` in a loop (until the result is `null`)?

Reply:
It does.

You mean don't register any callbacks on any events and straight up `read()` in a `while` loop? Wouldn't it make this method blocking (not async) and thus not ideal?

Not too opinionated as this is currently purely surfaced in anticipation for SDK users to use as convenience, DWN itself doesn't use this, so if you want me to change it I will, just let me know!

Reply:
i thought `read()` returned a `Promise`?

Reply:
I guess this is one (of many?) differences between browser-land and server-land when it comes to readable streams?! The abstraction that `readable-stream` provides adopts the node.js pattern, to which, wouldn't have worked with just a while loop based on their documentation anyway, still need to handle the `readable` event with the loop approach.

It's kinda a lose-lose situation in the sense that the web dev hates the SDK 'coz they can't use the web readable stream API directly, and I hate my job because I can only half-please everyone at best (thus please no one). But here we are, that's the requirement we got, at least it's cross platform.... I hope.